frontend/app/middleware/event-access-guard.ts

```diff
@@ -0,0 +1,28 @@
+export default defineNuxtRouteMiddleware(async (to) => {
+  const toast = useToast()
+  let description = 'Nieznany błąd'
+
+  try {
+    const { error } = await useApi('/event/status', {
+      redirect: 'error',
+    })
+
+    if (error.value?.data) {
+      const response = error.value.data as any
+      switch (response.code) {
+        case 403:
+          return showError({
+            statusCode: 403,
+            data: to,
+          })
+        case 410:
+          description = 'Nie możesz otworzyć tej strony po zakończeniu wydarzenia'
+          toast.add({ title: `Błąd ${response.code}`, description: `${description}`, color: 'error' })
+          return '/'
+      }
+    }
+  } catch {
+    toast.add({ title: 'Błąd', description: `${description}`, color: 'error' })
```

Comment:
`useFetch` never throws an error, errors from $fetch are caught and passed to the `error` property